### PR TITLE
Client number accepts 10 digit mobile number

### PR DESCRIPTION
**Pull request by piyushmhrzn**
What happened?
- Client Number field only accepts values upto 255. 

What should happen?
- Client Number field should only accept a 10 digit mobile number.
- Custom error messages for different errors

### CODE REVIEW

File: app/Http/Requests/ClientRequest.php [hunk 1]
```diff
@@ -25,9 +25,17 @@ public function rules()
...
         return [
             'client_name' => 'required|regex:/^[\pL\s\-]+$/u|max:255',
-            'client_number' => 'required|integer|max:255',
+            'client_number' => 'required|numeric|digits:10|regex:/^\d{10}$/',
             'client_email' => 'required | email |max:255|unique:clients',
             'status' => 'required | boolean',
         ];
     }
+
+    public function messages()
```
Comment (sahajmalla) [inline app/Http/Requests/ClientRequest.php:34]: I don't see the point of using the messages. Same message will be shown by laravel itself

File: app/Http/Requests/ClientRequest.php [hunk 2]
```diff
@@ -25,9 +25,17 @@ public function rules()
     {
         return [
             'client_name' => 'required|regex:/^[\pL\s\-]+$/u|max:255',
-            'client_number' => 'required|integer|max:255',
+            'client_number' => 'required|numeric|digits:10|regex:/^\d{10}$/',
```
Comment (sahajmalla) [inline app/Http/Requests/ClientRequest.php]: Add comments on what Regex is doing. This will be beneficial for future understanding.

**Review by sahajmalla — CHANGES_REQUESTED** [review]

File: app/Http/Requests/ClientRequest.php (hunk 1 above)
Comment (piyushmhrzn) [reply app/Http/Requests/ClientRequest.php:34]: It was just to add custom message like "It should be a valid mobile number".

File: app/Http/Requests/ClientRequest.php [hunk 3]
```diff
@@ -24,10 +24,13 @@ public function authorize()
     public function rules()
     {
         return [
-            'client_name' => 'required|regex:/^[\pL\s\-]+$/u|max:255',
-            'client_number' => 'required|integer|max:255',
+            // name-regex: allows for spaces and hyphens while also ensuring that the name only contains letters.
+            'client_name' => 'required|regex:/^[\pL\s\-]+$/u|max:255',  
+            // number-regex:  it contains exactly 10 digits
+            'client_number' => 'required|numeric|digits:10|regex:/^\d{10}$/',
```
Comment (sahajmalla) [inline app/Http/Requests/ClientRequest.php:30]: numeric and digits:10 already satisfies the regex. You can remove it and check.

**Review by sahajmalla — CHANGES_REQUESTED** [review]

File: app/Http/Requests/ClientRequest.php (hunk 3 above)
Comment (piyushmhrzn) [reply app/Http/Requests/ClientRequest.php]: done. regex not needed as numeric and digits:10 satisfies the condition

**Review by sahajmalla — APPROVED** [review]

**Pull request merged**